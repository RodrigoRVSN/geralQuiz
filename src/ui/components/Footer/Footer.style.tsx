import { experimentalStyled as styled } from "@material-ui/core/styles";
import { Container, Typography } from "@material-ui/core";

export const FooterStyled = styled("footer")`
  background-color: black;
  color: ${({ theme }) =>
    theme.palette.getContrastText(theme.palette.primary.main)};
  padding: ${({ theme }) => theme.spacing(2)} 0;
  margin-top: auto;
  text-align: center;
  &:hover {
    color: ${({ theme }) => theme.palette.text.primary};
  }
`;
