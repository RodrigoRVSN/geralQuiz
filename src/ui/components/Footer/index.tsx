import React from "react";
import { FooterStyled } from "./style";
import { Typography } from "@material-ui/core";

const Footer: React.FC = () => {
  return (
    <>
      <FooterStyled>
        <Typography variant={"body2"} sx={{ marginTop: 2 }}>
          <a data-testid="aboutLink" href="https://github.com/RodrigoRVSN" target="_blank">Desenvolvido por Rodrigo</a>
        </Typography>
      </FooterStyled>
    </>
  );
};

export default Footer;
