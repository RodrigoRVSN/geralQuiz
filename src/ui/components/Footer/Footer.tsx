import React from "react";
import { FooterStyled } from "./Footer.style";
import { Typography } from "@material-ui/core";

const Footer: React.FC = () => {
  return (
    <>
      <FooterStyled>
        <Typography variant={"body2"} sx={{ marginTop: 2 }}>
          <a href="https://github.com/RodrigoRVSN" target="_blank">Desenvolvido por Rodrigo</a>
        </Typography>
      </FooterStyled>
    </>
  );
};

export default Footer;
