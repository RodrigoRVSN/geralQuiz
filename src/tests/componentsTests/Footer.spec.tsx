import React from "react";
import { fireEvent, render, getByRole, screen} from "@testing-library/react";
import Footer from "../../ui/components/Footer";

describe("<Footer />", () => {

  it("should render the Footer and click my github", () => {
    const {getByTestId} = render(<Footer />, {});
    fireEvent.click(getByTestId("aboutLink"));
    expect(screen.getByRole('link',{name: 'Desenvolvido por Rodrigo'})).toBeInTheDocument();
  });

    it('should match snapshot', ()=>{
      const {container} = render(<Footer/>);
      expect(container).toMatchSnapshot();
    });

});
