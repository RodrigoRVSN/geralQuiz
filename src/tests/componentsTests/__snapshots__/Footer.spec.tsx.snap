// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<div>
  <footer
    class="css-o47hxc"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-11huu28-MuiTypography-root"
    >
      <a
        data-testid="aboutLink"
        href="https://github.com/RodrigoRVSN"
        target="_blank"
      >
        Desenvolvido por Rodrigo
      </a>
    </p>
  </footer>
</div>
`;
